### ISSUE TITLE
allow file Queue to (optionally) autosave on get()

### ISSUE THREAD

**Issue by AndyMan1** [peter-wangxu/persist-queue#116]
I'm trying out the file-based queue and it has what appears (to me) to be odd and unexpected behavior. Calling get() on the queue doesn't actually pop the item off the queue persistently. It seems you need to call task_done() after calling get() to save.

For example:
```
>>> from persistqueue import Queue
>>> q = Queue('/tmp/test')
>>> q.put(1)
>>> q.put(2)
>>> q.put(3)
>>> q.get()
1
>>> q.get()
2
>>> exit()
```
and then trying again:
```
>>> from persistqueue import Queue
>>> q = Queue('/tmp/test')
>>> q.get()
1
>>> q.get()
2
```
That doesn't seem intuitive or correct. I'd expect the retry attempt's first `get()` call to return `3`. The documentation for the Python standard lib Queue says "Remove and return an item from the queue." So remove should remove it, not sort-of-remove it. But you have to call `task_done()` to really remove it.

That doesn't seem like it matches what I'm reading as the purpose of `task_done()` in python's documentation which, if I'm interpresting it correctly, is basically to keep a counter going that allows you to call `join()` on the queue until it's empty. If I have no use for `join()`, I have no reason to call `task_done()`, right?

I think I understand where this is coming from. It allows you to retry if you don't finish the item or crash, and allows you to space out your disk writes for performance reasons if you need that.

So with all that in mind, would you be open to an optional init flag for the Queue to allow autosaving on a `get()` call? Perhaps mirroring the `auto_commit` flag/behavior of the `SQLiteQueue`? 

Autosaving would happen within the lock of the `get()`, keeping things atomic and persisting on every call. It would, at least in my mind, more closely match the standard lib's Queue behavior. Keeping it optional and off by default would keep things backwards compatible and people could ignore it if it's not their desired behavior. 

Would that be possible?

**Comment by peter-wangxu** [comment]
Agree with you.
The idea to implement `task_done` is to keep best compatibility with what's in the std lib.
If you want add the auto_commit flag, both `task_done` and `join` should be no-op with the enabled

I am fine to add this behavior and bear in mind the doc for this :)

**Comment by peter-wangxu** [comment]
fixed by #118
